conversion/src/main/java/com/qualinsight/mojo/cobertura/transformation/JaxbCoberturaToSonarQubeReportConverter.java

```diff
@@ -103,4 +120,31 @@ public void convertReport(
     }
 
   }
+
+  /**
+   * Find the correct file prefix path for the given sourceFileName, choose the first one if no file exists that matches
+   * the prefixes present in the project
+   *
+   * @param sonarSourcesPathPrefixMap
+   * @param sourceFileName
+   * @return
+   */
+  String findPrefix(final Map<String, String> sonarSourcesPathPrefixMap, final String sourceFileName){
+    String defaultPrefix = null;
+    File fileToCheck;
+    for(Map.Entry<String, String> sonarSourcesPathPrefixEntry : sonarSourcesPathPrefixMap.entrySet()){
+      fileToCheck = new File(sonarSourcesPathPrefixEntry.getKey() + sourceFileName);
+      if(fileToCheck.exists()){
+        return sonarSourcesPathPrefixEntry.getValue();
+      }else {
+        //Should we even bother defaulting?
```

Comment:
The entire defaulting thing here can potentially be removed and just returning a default "" and just know that the file name is not going to resolve because we don't know what it is.